plugins/overwatch.rb

```diff
@@ -0,0 +1,660 @@
+                  session_log(sid, "Could not add route to subnet #{subnet}/#{netmask} from #{interface.mac_name}")
+                end
+              rescue ::Exception => error
+                session_log(sid, "Could not add route to subnet #{subnet}/#{netmask} from #{interface.mac_name}")
+                session_log(sid, error.to_s)
+              end
+            end
+          end 
+        end
+      end
+    end
+
+    # Removes routes associated with stale sessions only if they are
+    # also present in sessions that are still active.
+    #
+    # @subnet [string class] route subnet
+    # @netmask [string array] route netmask
+    # @curr_sid [int class] Session ID of the current session being processed
+    # @stale_sids [int array] Array of session ID's that have been marked as stale.
+    #
+    # @return [void] A useful return value is not expected here
+    def remove_if_stale(subnet, netmask, curr_sid, stale_sids)
+      Rex::Socket::SwitchBoard.each do | route |
+        if stale_sids.include? route.comm.sid # See if a route is associated with a stale session
+          if route.subnet == subnet && route.netmask == netmask # See if that route matches the one currently being processed
+            session_log(route.comm.sid, "Deleting route #{route.subnet}/#{route.netmask}")
+            begin
+              Rex::Socket::SwitchBoard.remove_route(route.subnet, route.netmask, route.comm) # Remove so fresh matching route can be added
+            rescue ::Exception => error
+              session_log(route.comm.sid, "Could not remove route to subnet #{route.subnet}/#{route.netmask}")
+              session_log(route.comm.sid, error.to_s)
+            end
+            return
+          end
+        end
+      end
+    end
+
+    # Take an IP address and a netmask and return the appropreate subnet "Network"
+    #
+    # @ip_addr [string class] Input IPv4 Address
+    # @netmask [string class] Input IPv4 Netmask
+    #
+    # @return [string class] The subnet related to the IP address and netmask
+    # @return [nil class] Something is out of range
+    def get_subnet(ip_addr, netmask)
+      return nil if !validate_cmd(ip_addr, netmask) #make sure IP and netmask are valid
+
+      nets = ip_addr.split('.')
+      masks = netmask.split('.')
+      output = ""
+
+      (0..3).each do | index |
+        octet = get_subnet_octet(int_or_nil(nets[index]), int_or_nil(masks[index]))
+        return nil if !octet
+        output << octet.to_s
+        output << '.' if index < 3
+      end
+      return output
+    end
+
+    # Input an octet of an IPv4 address and the cooresponding octet of the
+    # IPv4 netmask then return the appropreate subnet octet.
+    #
+    # @net  [integer class] IPv4 address octet
+    # @mask [integer class] Ipv4 netmask octet
+    #
+    # @return [integer class] Octet of the subnet
+    # @return [nil class] If an input is nil
+    def get_subnet_octet(net, mask)
+      return nil if !net || !mask
```

Comment:
More nitpicky commentary - _return nil unless (net || mask)_ or if you prefer _return nil if !(net || mask)_